.github/pull_request_template

```diff
@@ -0,0 +1,37 @@
+# Pull Request Guidelines
+
+## Please consider splitting up big PRs (more than 10-15 files) into several small ones for easier reviews
+
+---
+
+# Pull Request Template
+
+## Issue=#<Number>
+
+## Summary (1-2 sentences)
+
+## Details
+
+### Challenges (what problems did I face?)(optional)
+
+### Alternative implementations (what alternatives were considered and why were they disregarded?) (optional)
+
+### Solution (how did I fix the problems?) (optional)
+
+### Does it work on mobile? (Screenshot for different screen sizes)
+
+### Implementation details (how was this change implemented?) (optional)
+
+Please delete options that are not relevant.
```

Comment:
We could use GitHub's labels instead and assign `Bug fix`, `Feature`, and `Breaking Change` labels to PRs.

In that case, we could remove this checklist from the template!